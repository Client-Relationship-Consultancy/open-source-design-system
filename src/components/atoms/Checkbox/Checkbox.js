import React from "react"
import styled from "styled-components"
import PropTypes from "prop-types"
import { colourPalette } from "brandColours"

const StyledCheckbox = styled.input.attrs({
  type: "checkbox",
})`
  appearance: none;
  width: 1rem;
  height: 1rem;
  border-radius: 20%;
  border: 2px solid ${props => props.theme.secondary.main.hex};
  outline: none;
  text-align: center;
  transition: 0.2s ease all;
  opacity: ${props => (props.disabled ? 0.4 : 1)};
  + label {
    color: ${props => props.theme.black.main.hex};
  }
  :checked {
    border: 2px solid ${props => props.theme.secondary.dark.hex};
    background-color: ${props => props.theme.white.main};
    + label {
      font-weight: bold;
      color: ${props => props.theme.secondary.dark.hex};
    }
    ::after {
      content: "\\2713";
      font-size: 0.8rem;
      line-height: 1rem;
      color: ${props => props.theme.secondary.dark.hex};
    }
  }
  :hover {
    cursor: ${props => (props.disabled ? "not-allowed" : "pointer")};
    + label {
      color: ${props => props.theme.secondary.dark.hex};
    }
  }
`
StyledCheckbox.defaultProps = {
  theme: colourPalette.examplePalette,
}
StyledCheckbox.displayName = "StyledCheckbox"

const Checkbox = props => {
  const { id, className, name, checked, disabled, onChange, theme } = props
  return (
    <StyledCheckbox
      id={id}
      className={className}
      name={name}
      checked={checked}
      disabled={disabled}
      onChange={onChange}
      theme={theme}
    />
  )
}

// Declare the display name of component for static build
// Display name should match export name
Checkbox.displayName = "Checkbox"

// Declare Checkbox propType values
Checkbox.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  name: PropTypes.string,
  checked: PropTypes.bool,
  disabled: PropTypes.bool,
  onChange: PropTypes.func,
  theme: PropTypes.object,
}

export default Checkbox
