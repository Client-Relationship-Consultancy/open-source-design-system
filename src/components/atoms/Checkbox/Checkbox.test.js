import React from "react"
import { mount } from "enzyme"
import "jest-styled-components"
import Checkbox from "./Checkbox"

describe("Input", () => {
  const component = mount(<Checkbox name="dogs" />)
  it("Input type should be checkbox", () => {
    expect(component.find("input").prop("type")).toEqual("checkbox")
  })
  it("Checkbox match previous snapshot", () => {
    expect(component).toMatchSnapshot()
  })
})
