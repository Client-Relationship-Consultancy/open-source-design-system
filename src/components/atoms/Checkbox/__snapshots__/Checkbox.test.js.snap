// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Checkbox match previous snapshot 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: 1rem;
  height: 1rem;
  border-radius: 20%;
  border: 2px solid #0B466B;
  outline: none;
  text-align: center;
  -webkit-transition: 0.2s ease all;
  transition: 0.2s ease all;
  opacity: 1;
}

.c0 + label {
  color: #272727;
}

.c0:checked {
  border: 2px solid #05263B;
}

.c0:checked + label {
  font-weight: bold;
  color: #05263B;
}

.c0:checked::after {
  content: "\\2713";
  font-size: 0.8rem;
  line-height: 1rem;
  color: #05263B;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover + label {
  color: #05263B;
}

<Checkbox
  name="dogs"
>
  <StyledCheckbox
    className="Checkbox__StyledCheckbox"
    name="dogs"
    theme={
      Object {
        "action": Object {
          "dark": Object {
            "RGB": "79,38,98",
            "hex": "#4E2662",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "173,101,207",
            "hex": "#AD65CF",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "116,60,143",
            "hex": "#743C8F",
            "on": "#FFFFFF",
          },
        },
        "background": Object {
          "RGB": "246,246,246",
          "hex": "#F6F6F6",
          "on": "#272727",
        },
        "black": Object {
          "main": Object {
            "RGB": "39,39,39",
            "hex": "#272727",
            "on": "#FFFFFF",
          },
          "tint20": Object {
            "RGB": "236,236,236",
            "hex": "#ECECEC",
            "on": "#272727",
          },
          "tint40": Object {
            "RGB": "199,199,199",
            "hex": "#C7C7C7",
            "on": "#272727",
          },
          "tint60": Object {
            "RGB": "160,160,160",
            "hex": "#A0A0A0",
            "on": "#FFFFFF",
          },
          "tint80": Object {
            "RGB": "105,105,105",
            "hex": "#696969",
            "on": "#FFFFFF",
          },
        },
        "complimentary": Object {
          "dark": Object {
            "RGB": "206,185,0",
            "hex": "#CEB900",
            "on": "#272727",
          },
          "light": Object {
            "RGB": "255,245,157",
            "hex": "#FFF59D",
            "on": "#272727",
          },
          "main": Object {
            "RGB": "255,238,82",
            "hex": "#FFEE52",
            "on": "#272727",
          },
        },
        "error": Object {
          "dark": Object {
            "RGB": "197,73,1",
            "hex": "#C54901",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "255,115,32",
            "hex": "#FF7320",
            "on": "#FFFFFF",
          },
        },
        "primary": Object {
          "dark": Object {
            "RGB": "102,22,20",
            "hex": "#661614",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "218,52,47",
            "hex": "#DA342F",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "164,40,36",
            "hex": "#A42824",
            "on": "#FFFFFF",
          },
        },
        "secondary": Object {
          "dark": Object {
            "RGB": "5,38,59",
            "hex": "#05263B",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "89,156,175",
            "hex": "#599CAF",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "11,70,107",
            "hex": "#0B466B",
            "on": "#FFFFFF",
          },
        },
        "surface": Object {
          "RGB": "255,255,255",
          "hex": "#FFFFFF",
          "on": "#272727",
        },
        "white": Object {
          "RGB": "255,255,255",
          "hex": "#FFFFFF",
          "on": "#272727",
        },
      }
    }
  >
    <StyledComponent
      className="Checkbox__StyledCheckbox"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "_foldedDefaultProps": Object {
            "theme": Object {
              "action": Object {
                "dark": Object {
                  "RGB": "79,38,98",
                  "hex": "#4E2662",
                  "on": "#FFFFFF",
                },
                "light": Object {
                  "RGB": "173,101,207",
                  "hex": "#AD65CF",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "116,60,143",
                  "hex": "#743C8F",
                  "on": "#FFFFFF",
                },
              },
              "background": Object {
                "RGB": "246,246,246",
                "hex": "#F6F6F6",
                "on": "#272727",
              },
              "black": Object {
                "main": Object {
                  "RGB": "39,39,39",
                  "hex": "#272727",
                  "on": "#FFFFFF",
                },
                "tint20": Object {
                  "RGB": "236,236,236",
                  "hex": "#ECECEC",
                  "on": "#272727",
                },
                "tint40": Object {
                  "RGB": "199,199,199",
                  "hex": "#C7C7C7",
                  "on": "#272727",
                },
                "tint60": Object {
                  "RGB": "160,160,160",
                  "hex": "#A0A0A0",
                  "on": "#FFFFFF",
                },
                "tint80": Object {
                  "RGB": "105,105,105",
                  "hex": "#696969",
                  "on": "#FFFFFF",
                },
              },
              "complimentary": Object {
                "dark": Object {
                  "RGB": "206,185,0",
                  "hex": "#CEB900",
                  "on": "#272727",
                },
                "light": Object {
                  "RGB": "255,245,157",
                  "hex": "#FFF59D",
                  "on": "#272727",
                },
                "main": Object {
                  "RGB": "255,238,82",
                  "hex": "#FFEE52",
                  "on": "#272727",
                },
              },
              "error": Object {
                "dark": Object {
                  "RGB": "197,73,1",
                  "hex": "#C54901",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "255,115,32",
                  "hex": "#FF7320",
                  "on": "#FFFFFF",
                },
              },
              "primary": Object {
                "dark": Object {
                  "RGB": "102,22,20",
                  "hex": "#661614",
                  "on": "#FFFFFF",
                },
                "light": Object {
                  "RGB": "218,52,47",
                  "hex": "#DA342F",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "164,40,36",
                  "hex": "#A42824",
                  "on": "#FFFFFF",
                },
              },
              "secondary": Object {
                "dark": Object {
                  "RGB": "5,38,59",
                  "hex": "#05263B",
                  "on": "#FFFFFF",
                },
                "light": Object {
                  "RGB": "89,156,175",
                  "hex": "#599CAF",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "11,70,107",
                  "hex": "#0B466B",
                  "on": "#FFFFFF",
                },
              },
              "surface": Object {
                "RGB": "255,255,255",
                "hex": "#FFFFFF",
                "on": "#272727",
              },
              "white": Object {
                "RGB": "255,255,255",
                "hex": "#FFFFFF",
                "on": "#272727",
              },
            },
          },
          "attrs": Array [
            Object {
              "type": "checkbox",
            },
          ],
          "componentStyle": ComponentStyle {
            "componentId": "Checkbox__StyledCheckbox-sc-8ptvpl-0",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "appearance:none;width:1rem;height:1rem;border-radius:20%;border:2px solid ",
              [Function],
              ";outline:none;text-align:center;transition:0.2s ease all;opacity:",
              [Function],
              ";+ label{color:",
              [Function],
              ";}:checked{border:2px solid ",
              [Function],
              ";background-color:",
              [Function],
              ";+ label{font-weight:bold;color:",
              [Function],
              ";}::after{content:\\"\\\\2713\\";font-size:0.8rem;line-height:1rem;color:",
              [Function],
              ";}}:hover{cursor:",
              [Function],
              ";+ label{color:",
              [Function],
              ";}}",
            ],
          },
          "displayName": "StyledCheckbox",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Checkbox__StyledCheckbox-sc-8ptvpl-0",
          "target": "input",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      name="dogs"
      theme={
        Object {
          "action": Object {
            "dark": Object {
              "RGB": "79,38,98",
              "hex": "#4E2662",
              "on": "#FFFFFF",
            },
            "light": Object {
              "RGB": "173,101,207",
              "hex": "#AD65CF",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "116,60,143",
              "hex": "#743C8F",
              "on": "#FFFFFF",
            },
          },
          "background": Object {
            "RGB": "246,246,246",
            "hex": "#F6F6F6",
            "on": "#272727",
          },
          "black": Object {
            "main": Object {
              "RGB": "39,39,39",
              "hex": "#272727",
              "on": "#FFFFFF",
            },
            "tint20": Object {
              "RGB": "236,236,236",
              "hex": "#ECECEC",
              "on": "#272727",
            },
            "tint40": Object {
              "RGB": "199,199,199",
              "hex": "#C7C7C7",
              "on": "#272727",
            },
            "tint60": Object {
              "RGB": "160,160,160",
              "hex": "#A0A0A0",
              "on": "#FFFFFF",
            },
            "tint80": Object {
              "RGB": "105,105,105",
              "hex": "#696969",
              "on": "#FFFFFF",
            },
          },
          "complimentary": Object {
            "dark": Object {
              "RGB": "206,185,0",
              "hex": "#CEB900",
              "on": "#272727",
            },
            "light": Object {
              "RGB": "255,245,157",
              "hex": "#FFF59D",
              "on": "#272727",
            },
            "main": Object {
              "RGB": "255,238,82",
              "hex": "#FFEE52",
              "on": "#272727",
            },
          },
          "error": Object {
            "dark": Object {
              "RGB": "197,73,1",
              "hex": "#C54901",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "255,115,32",
              "hex": "#FF7320",
              "on": "#FFFFFF",
            },
          },
          "primary": Object {
            "dark": Object {
              "RGB": "102,22,20",
              "hex": "#661614",
              "on": "#FFFFFF",
            },
            "light": Object {
              "RGB": "218,52,47",
              "hex": "#DA342F",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "164,40,36",
              "hex": "#A42824",
              "on": "#FFFFFF",
            },
          },
          "secondary": Object {
            "dark": Object {
              "RGB": "5,38,59",
              "hex": "#05263B",
              "on": "#FFFFFF",
            },
            "light": Object {
              "RGB": "89,156,175",
              "hex": "#599CAF",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "11,70,107",
              "hex": "#0B466B",
              "on": "#FFFFFF",
            },
          },
          "surface": Object {
            "RGB": "255,255,255",
            "hex": "#FFFFFF",
            "on": "#272727",
          },
          "white": Object {
            "RGB": "255,255,255",
            "hex": "#FFFFFF",
            "on": "#272727",
          },
        }
      }
    >
      <input
        className="c0 Checkbox__StyledCheckbox"
        name="dogs"
        type="checkbox"
      />
    </StyledComponent>
  </StyledCheckbox>
</Checkbox>
`;
